src/hotspot/share/runtime/globals.hpp

```diff
@@ -733,6 +733,10 @@ const int ObjectAlignmentInBytes = 8;
           "at one time (minimum is 1024).")                                 \
           range(1024, max_jint)                                             \
                                                                             \
+  product(intx, MonitorUnlinkBatch, 500, DIAGNOSTIC,                        \
+          "The maximum number of monitors to unlink in one batch.")         \
+          constraint(MonitorUnlinkBatchConstraintFunc, AfterErgo)           \
```

Comment:
Could this have a `range(1, max_jint)` as well? 
Should it mention MonitorDeflationMax? Or does it even need the constrained by MonitorDeflationMax. While it seems useless to have a value greater than MonitorDeflationMax, it does not seem invalid.

Reply:
Right. We can do the `range` check here, and simplify the constraint function. And actually we don't have to cap at `MonitorDeflationMax`. I am fixing this in newer commits.